src/lib.rs

```diff
@@ -45,17 +45,25 @@ impl Eta {
         Eta::create_instance(tasks_count, time_accuracy, tasks_done)
     }
 
-    /*fn pause(&self){
-        //tbd
+    pub fn pause(&mut self) {
+        if self.paused.is_none() {
+            self.paused = Some(Instant::now());
+        }
+    }
+
+    pub fn resume(&mut self) {
+        if !self.paused.is_none() {
+            self.recent_time = Instant::now();
+            self.paused = None;
+        }
     }
-    fn resume(&self){
-        //tbd
-    }*/
 
     pub fn step(&mut self) {
```

Comment:
Please allow `step()` to happen here by removing the main if and rather make adjustments to `Eta.paused` i.e. `self.paused = (None, 0)`